testsuite/src/test/java/org/jboss/da/test/server/ReportsWithWhitelistedArtifactsRemoteTest.java

```diff
@@ -0,0 +1,492 @@
+
+        GAVRequest gavRequest = getDefaultGAVRequest();
+
+        // Query all white artifacts from Product 1 version 2
+        gavRequest.getProductVersionIds().add(prodVer1_2.getId());
+
+        ArtifactReport artifactReport = reportGenerator.getReport(gavRequest);
+
+        // Ensure we got an artifact report back 
+        assertTrue(artifactReport != null);
+
+        // Dependencies from Product 1 version 2 expected
+        assertFalse(artifactReport.getAvailableVersions().contains(whiteArtifact1_1_2.getVersion()));
+        assertTrue(artifactReport.getAvailableVersions().contains(whiteArtifact1_2_2.getVersion()));
+        assertFalse(artifactReport.getAvailableVersions().contains(whiteArtifact2_1_c.getVersion()));
+    }
+
+    private GAV artifactToGAV(RestProductArtifact art) {
+        return new GAV(art.getGroupId(), art.getArtifactId(), art.getVersion());
+    }
+
+    private LookupGAVsRequest getDefaultLookupRequest() {
+        Set<String> productNames = new HashSet<>();
+        Set<Long> productVersionIds = new HashSet<>();
+
+        List<GAV> gavs = new ArrayList<GAV>();
+        gavs.add(hibernateGav);
+        gavs.add(daGav);
+        gavs.add(jacksonGav);
+
+        LookupGAVsRequest req = new LookupGAVsRequest(productNames, productVersionIds, gavs);
+
+        return req;
+    }
+
+    @Test
+    public void testGavsLookupAllWhiteArtifactsPresent() throws CommunicationException,
+            FindGAVDependencyException {
+
+        LookupGAVsRequest req = getDefaultLookupRequest();
+
+        // Query all white artifacts from all products
+
+        List<LookupReport> lookupReports = reportGenerator.getLookupReportsForGavs(req);
+
+        assertTrue(lookupReports != null);
+
+        // Expecting to see all whitelist versions included
+        for (LookupReport lr : lookupReports) {
+            if (hibernateGav.equals(lr.getGav())) {
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact1_1_2.getVersion()));
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact1_2_2.getVersion()));
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact2_1_c.getVersion()));
+                assertTrue(lr.getBestMatchVersion().equals(hibernateGav.getVersion()));
+            } else if (daGav.equals(lr.getGav())) {
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact1_1_1.getVersion()));
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact1_2_1.getVersion()));
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact2_1_a.getVersion()));
+                assertTrue(lr.getBestMatchVersion().equals(daGav.getVersion()));
+            } else if (jacksonGav.equals(lr.getGav())) {
+                assertTrue(lr.getAvailableVersions().contains(whiteArtifact2_1_b.getVersion()));
+                assertTrue(lr.getBestMatchVersion().equals(jacksonGav.getVersion()));
+            }
```

Comment:
Do we expect that all the 3 ifs are visited? Now, if lookupReports is empty or doesn't contain ``hibernateGav``, ``daGav`` or ``jacksonGav`` we don't visit the condition branch. Is it ok?

You may add a counter and increment it in each branch and after the if check if the counter is 3.